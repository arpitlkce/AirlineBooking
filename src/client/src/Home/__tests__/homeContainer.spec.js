import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import HomeContainer from '../homeContainer';
const mockStore = configureMockStore();
const store = mockStore({});

describe('Home Container Component', () => {
    it('should render correctly in  mode', () => {
        const component = renderer.create(<Provider store={store}><HomeContainer /></Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
