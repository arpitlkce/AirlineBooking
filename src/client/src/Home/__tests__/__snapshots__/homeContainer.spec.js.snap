// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container Component should render correctly in  mode 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Go Fly
    </a>
    <ul
      className="mar-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <h3>
           Welcome to Go Fly 
        </h3>
      </li>
    </ul>
    <button
      aria-label={null}
      className="btn btn-light"
      onClick={[Function]}
      type="button"
    >
      Log Out
    </button>
  </nav>
  <div
    className="container"
  >
    <ul
      className="nav nav-tabs"
    >
      <li
        className="nav-item"
      >
        <a
          className="active nav-link"
          onClick={[Function]}
        >
          New Booking
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          onClick={[Function]}
        >
          Check Status
        </a>
      </li>
    </ul>
    <div
      className="tab-content"
    >
      <div
        className="tab-pane active"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-12"
          >
            <div
              className="container"
            >
              <div
                className="mar-top-20 mar-bottom-20 row"
              >
                <div
                  className="col-md-6"
                >
                  <div
                    className="btn-group"
                    role="group"
                  >
                    <button
                      aria-label={null}
                      className="activeButton btn btn-outline-primary"
                      name="journeyType"
                      onClick={[Function]}
                      type="button"
                      value="single"
                    >
                      One Way
                    </button>
                    <button
                      aria-label={null}
                      className="btn btn-outline-primary"
                      name="journeyType"
                      onClick={[Function]}
                      type="button"
                      value="return"
                    >
                      Round Trip
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <label
                    className=""
                    htmlFor="boarding"
                  >
                    Boarding
                  </label>
                  <div
                    className="Dropdown-root"
                  >
                    <div
                      aria-haspopup="listbox"
                      className="Dropdown-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="Dropdown-placeholder"
                      >
                        Select an option
                      </div>
                      <div
                        className="Dropdown-arrow-wrapper"
                      >
                        <span
                          className="Dropdown-arrow"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <label
                    className=""
                    htmlFor="destination"
                  >
                    Destination
                  </label>
                  <div
                    className="Dropdown-root"
                  >
                    <div
                      aria-haspopup="listbox"
                      className="Dropdown-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="Dropdown-placeholder"
                      >
                        Select an option
                      </div>
                      <div
                        className="Dropdown-arrow-wrapper"
                      >
                        <span
                          className="Dropdown-arrow"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="marginInfo row"
              >
                <div
                  className="col-md-6"
                >
                  <label
                    className=""
                    htmlFor="departureDate"
                  >
                    Departure
                  </label>
                  <div>
                    <div
                      className="react-datepicker-wrapper"
                    >
                      <div
                        className="react-datepicker__input-container"
                      >
                        <input
                          className=""
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          readOnly={false}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <label
                    className=""
                    htmlFor="returnDate"
                  >
                    Return
                  </label>
                  <div>
                    <div
                      className="react-datepicker-wrapper"
                    >
                      <div
                        className="react-datepicker__input-container"
                      >
                        <input
                          className=""
                          disabled={true}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          readOnly={false}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <label
                    className=""
                    htmlFor="adult"
                  >
                    Adult
                  </label>
                  <input
                    className="form-control"
                    id="adult"
                    max="6"
                    min="1"
                    name="adult"
                    onChange={[Function]}
                    type="number"
                  />
                </div>
                <div
                  className="col-md-4"
                >
                  <label
                    className=""
                    htmlFor="children"
                  >
                    Children
                  </label>
                  <input
                    className="form-control"
                    id="children"
                    min="0"
                    name="children"
                    onChange={[Function]}
                    type="number"
                  />
                </div>
                <div
                  className="col-md-2"
                >
                  <label
                    className=""
                    htmlFor="infant"
                  >
                    Infant
                  </label>
                  <input
                    className="form-control"
                    id="infant"
                    min="0"
                    name="infant"
                    onChange={[Function]}
                    type="number"
                    value={0}
                  />
                </div>
              </div>
              <div
                className="mar-top-20 mar-bottom-20 row"
              >
                <div
                  className="col-md-6"
                >
                  <div
                    className="btn-group"
                    role="group"
                  >
                    <button
                      aria-label={null}
                      className="activeButton btn btn-outline-primary"
                      name="flightClass"
                      onClick={[Function]}
                      type="button"
                      value="economy"
                    >
                      Economy
                    </button>
                    <button
                      aria-label={null}
                      className="btn btn-outline-primary"
                      name="flightClass"
                      onClick={[Function]}
                      type="button"
                      value="business"
                    >
                      Business
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <button
                    aria-label={null}
                    className="btn btn-success disabled"
                    disabled={true}
                    onClick={[Function]}
                    type="button"
                  >
                    Search
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="tab-pane"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="container"
            >
              <div
                className="mar-top-20 mar-bottom-20 row"
              >
                <div
                  className="col"
                >
                  <label
                    className="mr-sm-2"
                    htmlFor="bookingId"
                  >
                    Booking ID
                  </label>
                  <input
                    className="form-control"
                    id="bookingId"
                    name="bookingId"
                    onChange={[Function]}
                    type="text"
                  />
                </div>
                <div
                  className="col"
                >
                  <button
                    aria-label={null}
                    className="mar-top-30 btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Submit
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
